content/blog/model-context-protocol-mcp-the-protocol-that-powers-ai-agents.md

```diff
@@ -0,0 +1,194 @@
+---
+title: "Part 3: Model Context Protocol (MCP): The Protocol That Powers AI Agents"
+date: 2025-07-18T14:23:55.595Z
+author: Dinesh R Singh
+authorimage: /img/dinesh-192-192.jpg
+disable: false
+tags:
+  - Agentic AI
+  - Gen AI
+  - Qdrant
+  - MCP
+  - Communication Protocol
+  - "Model Context Protocol "
+---
+<style>
+li {
+   font-size: 27px;
+   line-height: 33px;
+   max-width: none;
+}
+</style>
+
+As AI agents grow beyond text generation into autonomous problem-solvers, a new challenge emerges — communication. Not between humans and AI, but between AI and the vast world of services, APIs, databases, and tools. That’s where **MCP (Model Context Protocol)** steps in.
+
+Inspired by [my post on medium](https://dineshr1493.medium.com/all-you-need-to-know-about-the-evolution-of-generative-ai-to-agentic-ai-part-3-mcp-model-context-f026578ff0dd), this blog demystifies the MCP standard — reinterpreted with clarity, depth, and real-world relevance to help you understand how AI agents actually get things done.
+If LLMs are the brains, MCP is the nervous system connecting them to the real world. Let’s unpack how this protocol makes agentic AI functional, contextual, and enterprise-ready.
+
+<center><img src="/img/mcp1.png" width="600" height="550" alt="MCP Arch" title="MCP Arch"></center>
+
+## What is MCP, and why does it matter?
+
+At its core, MCP is a standardized way for AI agents to communicate with external services. Instead of treating each tool or database as a black box, MCP defines a consistent interface — allowing the agent to send structured requests and receive contextual responses.
+
+Imagine an agent saying:
+
+“Here’s the context, here’s what I need — now act smartly based on it.”
+
+That’s the essence of MCP. It removes ambiguity, reduces dependency on ad hoc code, and enables agents to **perform tasks with understanding, not just commands.**
+
+<center><img src="/img/mcp2.png" width="600" height="550" alt="MCP Flow" title="MCP Flow"></center>
+
+## The building blocks of MCP
+
+MCP is composed of three major components:
+
+* MCP Client: Resides inside the AI agent and is responsible for making requests.
+* MCP Server: Wraps around external tools or services and handles incoming requests.
+* MCP Protocol: Uses JSON-RPC over transport layers like:
```

Comment:
In the preview here (https://deploy-preview-3099--hpe-dev-portal.netlify.app/blog/model-context-protocol-mcp-the-protocol-that-powers-ai-agents/) there is a line space between the 
MCP Server: Wraps around external tools or services and handles incoming requests.

and

MCP Protocol: Uses JSON-RPC over transport layers like: